Examples/ObjectiveC/ClusteringWithImagesExample.m

```diff
@@ -0,0 +1,69 @@
+//
+//  ClusteringWithImagesExample.m
+//  Examples
+//
+//  Created by Joshua Erb on 7/30/18.
+//  Copyright © 2018 Mapbox. All rights reserved.
+//
+
+#import "ClusteringWithImagesExample.h"
+@import Mapbox;
+
+NSString *const MBXExampleClusteringWithImages = @"ClusteringWithImagesExample";
+
+@interface ClusteringWithImagesExample () <MGLMapViewDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) UIImage *icon;
+
+@end
+
+@implementation ClusteringWithImagesExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    // Initialize the Map
+    self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds
+                                            styleURL:[MGLStyle lightStyleURL]];
+    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    self.mapView.delegate = self;
+    [self.view addSubview:self.mapView];
+    
+    self.icon = [UIImage imageNamed:@"circle"];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    NSURL *url = [NSURL fileURLWithPath:[[NSBundle mainBundle] pathForResource:@"ports" ofType:@"geojson"]];
+    
+    // Retrieve data and set as style layer source
```

Comment:
Can add more comments on sources/style layers here and in the lines below? It might be useful to mention the source data is used to load data into the map but style layers actually configure visual styling to the data. It might be also good to clarify that three different style layers are required for this whole thing: a layer to display unclustered points, a layer to display clustered images, and a layer to display labels for the clusters.

Side note, can you include the unclustered points too, like we do in the original clustering example?

```objc
    // Show unclustered features as icons. The `cluster` attribute is built into clustering-enabled source features.
    MGLSymbolStyleLayer *ports = [[MGLSymbolStyleLayer alloc] initWithIdentifier:@"ports" source:source];
    ports.iconImageName = [NSExpression expressionForConstantValue:@"icon"];
    ports.iconColor = [NSExpression expressionForConstantValue:[[UIColor darkGrayColor] colorWithAlphaComponent:0.9]];
    ports.predicate = [NSPredicate predicateWithFormat:@"cluster != YES"];
    [style addLayer:ports];
```

Reply:
Just want to note here that because this example is using `MGLSymbolStyleLayer` for the clustering, only 2  different style layers are necessary: one for labels and images and one for unclustered images. 

You need three layers if you're using a CircleStyleLayer (which doesn't allow you to set a `.text` _and_ an `.iconImageName`.